File: Objective-C/Translator/XmlParser.m
//
//  XmlParser.m
//  MyAWS
//
//  Created by Adam Nagy on 06/11/2012.
//  Copyright (c) 2012 Adam Nagy. All rights reserved.
//

#import "XmlParser.h"

@implementation XmlParser

@synthesize translation;

static bool translationTag = false;

- (void)parser:(NSXMLParser *)parser 
               didStartElement:(NSString *)elementName 
               namespaceURI:(NSString *)namespaceURI 
               qualifiedName:(NSString *)qualifiedName 
	       attributes:(NSDictionary *)attributeDict {
	
  if ([elementName isEqualToString:@"translation"])
    translationTag = true;
}

- (void)parser:(NSXMLParser *)parser foundCharacters:(NSString *)string {
  
  if (translationTag)
  {
    if (!translation)
      translation = [NSString stringWithFormat:@""];
    
    translation =  [translation stringByAppendingString:string];
  }
}

- (void)parser:(NSXMLParser *)parser 
               didEndElement:(NSString *)elementName
               namespaceURI:(NSString *)namespaceURI 
               qualifiedName:(NSString *)qName {

  if ([elementName isEqualToString:@"translation"])
    translationTag = false;
}

@end

File: Swift/Translator/XmlParser.swift
//
//  XmlParser.swift
//  Translator
//
//  Created by Adam Nagy on 09/11/2015.
//  Copyright © 2015 Adam Nagy. All rights reserved.
//

import Foundation

class XmlParser: NSXMLParser,
  NSXMLParserDelegate {
  
  var translationTag: Bool = false
  var translation: String = ""
  
  func parser( parser: NSXMLParser, didStartElement elementName: String, namespaceURI: String?, qualifiedName qName: String?, attributes attributeDict: [String : String]) {
    if elementName == "translation" {
      translationTag = true
    }
  }
  
  func parser(parser: NSXMLParser, foundCharacters string: String) {
    if translationTag {
      translation = translation + string
    }
  }
  
  func parser(parser: NSXMLParser,
    didEndElement elementName: String,
    namespaceURI: String?,
    qualifiedName qName: String?) {
      if elementName == "translation" {
        translationTag = false
      }
  }
}


